### DISCUSSION TITLE
Conditional action execution

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#3275] [Ideas]
Hi

I think budibase should support to add conditional action execution so that not all action step will be executed in sequence.

For example I only wanted to execute "Save Row" action only if "Execute Query" return certain value etc.

![image](https://user-images.githubusercontent.com/597759/140646115-791ba592-f4a0-4735-969b-ebebc22dec97.png)

**Comment by mjashanks** [comment]
I agree that this would be really good to have. However, there could be a lot of work in it. It's basically the `Automations` section, client side - and I don't think that there would be anything re-usable from the existing `Automations` section :)

Someday I would like this, but I don't see it happening in the next few (6?) months. Thanks for the suggestion though - I'm happy for others to upvote this to show some interest.

**Reply by ryzam** [reply]
I think this feature quite important. My idea is to add checkbox inform that execution of current step only if previous step/action successfully.

**Reply by mjashanks** [reply]
We made some improvements on this. Still no conditional execution.

However, you can now use values from previous steps... e.g. `{{ Action 1.Saved Row._id }}`

**Comment by gzqn** [comment]
conditional execution would be good

**Reply by ryzam** [reply]
March release have introduced conditional steps .
